validator/sawtooth_validator/execution/context_manager.py

```diff
@@ -421,48 +607,120 @@ def run(self):
             c_id, inflated_address_list = context_id_list_tuple
             inflated_value_map = {k: v for k, v in inflated_address_list}
             if c_id in self._contexts:
-                self._contexts[c_id].set_futures(inflated_value_map,
-                                                 from_tree=True)
+                self._contexts[c_id].set_from_tree(inflated_value_map)
 
 
 class _ContextFuture(object):
-    def __init__(self, address, wait_for_tree=False):
+    """Controls access to bytes set in the _result variable. The booleans
+     that are flipped in set_result, based on whether the value is being set
+     from the merkle tree or a direct set on the context manager are needed
+     to later determine whether the value was set in that context or was
+     looked up as a new address location from the merkle tree and then only
+     read from, not set.
+
+    In any context the lifecycle of a _ContextFuture can be several paths:
```

Comment:
In general, what determines which path it takes? The diagrams below are good, but are sparse on explanation.